mos-platform/neo6502/api/console.c

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 LLVM-MOS Project
+// Licensed under the Apache License, Version 2.0 with LLVM Exceptions.
+// See https://github.com/llvm-mos/llvm-mos-sdk/blob/main/LICENSE for license
+// information.
+
+#include <string.h>
+#include "../neo6502.h"
+#include "../kernel.h"
+#include "api-internal.h"
+
+__attribute__((leaf))
+char neo_console_read_char(void) {
+    KSendMessageSync(API_GROUP_CONSOLE, API_FN_READ_CHAR);
+    return ControlPort.params[0];
+}
+
+__attribute__((leaf))
+uint8_t neo_console_status(void) {
+    KSendMessageSync(API_GROUP_CONSOLE, API_FN_CONSOLE_STATUS);
+    return ControlPort.params[0];
+}
+
+__attribute__((leaf))
+void neo_console_read_line_p(void *line) {
+    *((volatile uint16_t*) (ControlPort.params)) = (uint16_t) line;
+    KSendMessageSync(API_GROUP_CONSOLE, API_FN_READ_LINE);
+}
+
+__attribute__((leaf))
+void neo_console_read_line(char *line) {
+    line[0] = 0;
+    neo_console_read_line_p(line);
+    __neo_depascalize_output(line);
+}
+
+__attribute__((leaf))
+void neo_console_define_hotkey_p(uint8_t hotkey, const void *str) {
```

Comment:
Was there a reason that the strings in the neo6502 APIs were declared `void*` instead of `char*`?

Reply:
I wanted to quickly distinguish the ones which take Pascal string types from the ones which take C string types. Not sure if I should just have them as `const char *` or if I should define a `struct neo_pstring` or...